File: lab1_Csharp/PartialCalculator.cs
namespace lab1_Csharp
{
    public class PartialCalculator
    {
        private int[] _array;
        private int _startIndex;
        private int _step;

        public long Sum { get; private set; }

        public PartialCalculator(int[] array, int startIndex, int step)
        {
            _array = array;
            _startIndex = startIndex;
            _step = step;
        }

        public void Calculate()
        {
            for (int i = _startIndex; i < _array.Length; i += _step)
            {
                Sum += _array[i];
            }
        }
    }
}

File: lab1_Java/src/PartialCalculator.java
public class PartialCalculator extends Thread
{
    private Calculator _calculator;
    private int[] _array;
    private int _startIndex;
    private int _step;

    public PartialCalculator(Calculator calculator, int[] array, int startIndex, int step)
    {
        _calculator = calculator;
        _array = array;
        _startIndex = startIndex;
        _step = step;
    }

    @Override
    public void run()
    {
        long sum = 0;
        for (int i = _startIndex; i < _array.length; i+= _step)
        {
            sum += _array[i];
        }
        _calculator.addSum(sum);
    }
}